File: output/ChatGPT/codenet/C++/Java/s611392782.java
// ACM-ICPCアジア地区予選2014 B. Miscalculation

import java.util.Scanner;

public class s611392782 {

    public static int mul(String str, int[] idx) {
        int res = str.charAt(idx[0]) - '0';
        while (idx[0] + 1 < str.length() && str.charAt(idx[0] + 1) == '*') {
            idx[0] += 2;
            res *= str.charAt(idx[0]) - '0';
        }
        return res;
    }

    public static int add(String str, int[] idx) {
        int res = mul(str, idx);
        while (idx[0] + 1 < str.length() && str.charAt(idx[0] + 1) == '+') {
            idx[0] += 2;
            res += mul(str, idx);
        }
        return res;
    }

    public static int bob(String str, int[] idx) {
        int res = str.charAt(idx[0]) - '0';
        while (idx[0] + 1 < str.length()) {
            if (str.charAt(idx[0] + 1) == '+') {
                res += str.charAt(idx[0] + 2) - '0';
            } else {
                res *= str.charAt(idx[0] + 2) - '0';
            }
            idx[0] += 2;
        }
        return res;
    }

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        while (input.hasNext()) {
            String s = input.next();
            int d = input.nextInt();
            int[] idx = {0};
            int a = add(s, idx);
            idx[0] = 0;
            int b = bob(s, idx);
            if (a == d) {
                System.out.println((b == d ? "U" : "M"));
            } else {
                System.out.println((b == d ? "L" : "I"));
            }
        }
    }
}

File: output/ChatGPT/codenet/C++/Python/s611392782.py
def mul(s, idx):
    res = int(s[idx])
    while idx + 1 < len(s) and s[idx + 1] == '*':
        idx += 2
        res *= int(s[idx])
    return res

def add(s, idx):
    res = mul(s, idx)
    while idx + 1 < len(s) and s[idx + 1] == '+':
        idx += 2
        res += mul(s, idx)
    return res

def bob(s, idx):
    res = int(s[idx])
    while idx + 1 < len(s):
        if s[idx + 1] == '+':
            res += int(s[idx + 2])
        else:
            res *= int(s[idx + 2])
        idx += 2
    return res

while True:
    try:
        s, d = input().split()
        idx = 0
        a = add(s, idx)
        idx = 0
        b = bob(s, idx)
        if a == int(d):
            print("U" if b == int(d) else "M")
        else:
            print("L" if b == int(d) else "I")
    except EOFError:
        break